downstream/modules/platform/proc-controller-create-credential-type.adoc

```diff
@@ -0,0 +1,214 @@
+====
+
+. In the *Input Configuration* field, specify an input schema which defines a set of ordered fields for that type. 
+The format can be in YAML or JSON, as shown:
++
+*YAML*
++
+[literal, options="nowrap" subs="+attributes"]
+----
+fields:
+  - type: string
+    id: username
+    label: Username
+  - type: string
+    id: password
+    label: Password
+    secret: true
+required:
+  - username
+  - password 
+----
++
+View more YAML examples at the link:https://yaml.org/spec/1.2.2/[YAML page]
++
+*JSON*
++
+[literal, options="nowrap" subs="+attributes"]
+----
+{
+"fields": [
+  {
+  "type": "string",
+  "id": "username",
+  "label": "Username"
+  },
+  {
+  "secret": true,
+  "type": "string",
+  "id": "password",
+  "label": "Password"
+   }
+  ],
+ "required": ["username", "password"]
+}
+----
++
+View more JSON examples at link:https://www.json.org/json-en.html[The JSON website]
++
+The configuration in JSON format below show each field and how they are used:
++
+[literal, options="nowrap" subs="+attributes"]
+----
+{
+  "fields": [{
+    "id": "api_token",               # required - a unique name used to
+                                     # reference the field value
+
+    "label": "API Token",            # required - a unique label for the
+                                     # field
+
+    "help_text": "User-facing short text describing the field.",
+
+    "type": ("string" | "boolean")   # defaults to 'string'
+
+    "choices": ["A", "B", "C"]       # (only applicable to `type=string`)
+
+    "format": "ssh_private_key"      # optional, can be used to enforce data
+                                     # format validity for SSH private key
+                                     # data (only applicable to `type=string`)
+
+    "secret": true,                  # if true, the field value will be encrypted
+
+    "multiline": false               # if true, the field should be rendered
+                                     # as multi-line for input entry
+                                     # (only applicable to `type=string`)
+},{
+    # field 2...
+},{
+    # field 3...
+}],
+
+"required": ["api_token"]            # optional; one or more fields can be marked as required
+},
+----
++
+When `type=string`, fields can optionally specify multiple choice options:
++
+[literal, options="nowrap" subs="+attributes"]
+----
+{
+  "fields": [{
+      "id": "api_token",          # required - a unique name used to reference the field value
+      "label": "API Token",       # required - a unique label for the field
+      "type": "string",
+      "choices": ["A", "B", "C"]
+  }]
+},
+----
+
+. In the *Injector Configuration* field, enter environment variables or extra variables that specify the values a credential type can inject.
+The format can be in YAML or JSON (see examples in the previous step).
++
+The configuration in JSON format below show each field and how they are used:
```

Comment:
Suggested re-write:

The following configuration in JSON format shows each field and how they are used: